routes/api/v1/favorites.js

```diff
@@ -0,0 +1,91 @@
+var express = require('express');
+var router = express.Router();
+var fetch = require('node-fetch');
+var User = require('../../../models').User;
+var Location = require('../../../models').Location;
+var Favorite = require('../../../models').Favorite;
+
+/* POST new favorite */
+router.post('/', function(req, res) {
+  var foundUser;
+  var locationName;
+  User.findOne({
+    where: {
+      api_key: req.body.api_key
+    },
+    include: 'locations'
+  })
+  .then(user => {
+    return new Promise((resolve, reject) => {
+      user ? resolve(user) : reject("Invalid API Key")
+    })
+  })
+  .then(user => {
+    return new Promise((resolve, reject) => {
+      location = user.locations.find(function(element) {
+        return element.name == req.body.location;
+      })
+
+      location ?
+      reject({error: req.body.location + " is already favorited"}) :
+      resolve(user)
+    })
+  })
+  .then(user => {
+    foundUser = user;
+    return findOrCreateCity(req.body.location);
```

Comment:
like that you broke this fetch call function out

Reply:
Thanks, I'll try and move more stuff out.